golang/trace/zipkin_tracereporter.go

```diff
@@ -0,0 +1,100 @@
+package trace
+
+import (
+	"encoding/base64"
+	"encoding/binary"
+	"net"
+	"time"
+
+	tc "github.com/uber/tchannel/golang"
+	"github.com/uber/tchannel/golang/thrift"
+	"github.com/uber/tchannel/golang/trace/thrift/gen-go/tcollector"
+)
+
+// ZipkinTraceReporter is a trace reporter that submits trace spans in to zipkin trace server.
+type ZipkinTraceReporter struct {
+	tchannel *tc.Channel
+}
+
+// Report method will submit trace span to tcollector server.
+func (r *ZipkinTraceReporter) Report(
+	span tc.Span, annotations []tc.Annotation, binaryAnnotations []tc.BinaryAnnotation, name string, endpoint *tc.Endpoint) {
+
+	thriftClient := thrift.NewClient(r.tchannel, "tcollector", nil)
+	client := tcollector.NewTChanTCollectorClient(thriftClient)
+
+	ctx, cancel := tc.NewContextBuilder(time.Second).
+		SetShardKey(Base64Encode(span.TraceID())).Build()
+	defer cancel()
+
+	thriftSpan := BuildZipkinSpan(span, annotations, binaryAnnotations, name, endpoint)
+	// client submit
+	client.Submit(ctx, thriftSpan)
+}
+
+// BuildZipkinSpan builds zipkin span based on tchannel span.
+func BuildZipkinSpan(span tc.Span, annotations []tc.Annotation, binaryAnnotations []tc.BinaryAnnotation, name string, endpoint *tc.Endpoint) *tcollector.Span {
```

Comment:
Does this method need to be exported? Your `_test.go` file can access unexported methods from the same package, so you could unexport this but still use it and test it.